mlrun/api/db/sqldb/db.py

```diff
@@ -2438,59 +3294,125 @@ def store_feature_set(
         versioned=True,
         always_overwrite=False,
     ) -> str:
+        return self._store_tagged_object(
+            session,
+            FeatureSet,
+            project,
+            name,
+            feature_set,
+            tag=tag,
+            uid=uid,
+            versioned=versioned,
+            always_overwrite=always_overwrite,
+        )
+
+    def _store_tagged_object(
+        self,
+        session,
+        cls,
+        project,
+        name,
+        tagged_object: typing.Union[
+            mlrun.common.schemas.FeatureVector,
+            mlrun.common.schemas.FeatureSet,
+        ],
```

Comment:
why is tagged_object wither FeatureVector or FeatureSet? isn't it supposed to be generic?

Reply:
Well, it is currently used by these 2 only.
I ended up not using it for artifacts as they required a more specific handling.
I left the function as a common anyway because it still made since.

Do you think I should just lose the type hinting?